Controller/RecommendationController.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * File containing the RecommendationController class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ * @version //autogentag//
+ */
+
+namespace EzSystems\RecommendationBundle\Controller;
+
+use eZ\Bundle\EzPublishCoreBundle\Controller;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use EzSystems\RecommendationBundle\Client\YooChooseNotifier;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
+use eZ\Publish\API\Repository\Values\Content\LocationQuery;
+use eZ\Publish\API\Repository\Values\Content\Query;
+use eZ\Publish\API\Repository\Values\Content\Query\SortClause;
+
+class RecommendationController extends Controller
+{
+    public function recommendationsAction( Request $request )
+    {
+        if ( !$request->isXmlHttpRequest() ) {
+            throw new BadRequestHttpException();
+        }
+
+        $userId = $this->getRepository()->getCurrentUser()->id;
+        $locationId = $request->query->get( 'locationId' );
+        $limit = $request->query->get( 'limit' );
+        $scenarioId = $request->query->get( 'scenarioId' );
+
+        $notifier = $this->get( 'ez_recommendation.client.yoochoose_notifier' );
+
+        $responseRecommendations = $notifier->getRecommendations(
+            $userId, $scenarioId, $locationId, $limit
+        );
+
+        $recommendedContentIds = array();
+
+        foreach ( $responseRecommendations[ 'recommendationResponseList' ] as $recommendation ) {
+            $recommendedContentIds[] = $recommendation[ 'itemId' ];
+        }
+
+        $content = array();
+
+        if ( count( $recommendedContentIds ) > 0 ) {
+
+            $criterion = new Criterion\LogicalAnd( array(
```

Comment:
I suggest you move the whole Content Query part to a private method (ideally, an external service, but a private method is a good step forward).

It would take care of the mapping above (since you don't need to map to count items, as seen in other comment), build the Query Object, and return `$searchResults->searchHits`.

It's not a must of course, but it would make things much cleaner.